packages/ember/tests/helpers/link_to_test/link_to_with_query_params_test.js

```diff
@@ -1,506 +1,567 @@
+      {{#link-to 'index' id='the-link'}}Index{{/link-to}}
+    `);
+
+    this.visit('/').then(() => {
+      this.$('#the-link').click();
+      let indexController = this.applicationInstance.lookup('controller:index');
+
+      deepEqual(indexController.getProperties('foo', 'bar'),
+        this.indexProperties,
+        'controller QP properties do not update'
+      );
     });
-  },
+  }
 
-  teardown: sharedTeardown
-});
+  ['@test doesn\'t update controller QP properties on current route when invoked (empty query-params obj)']() {
+    this.addTemplate('index', `
+      {{#link-to 'index' (query-params) id='the-link'}}Index{{/link-to}}
+    `);
 
-QUnit.test('doesn\'t update controller QP properties on current route when invoked', function() {
-  setTemplate('index', compile(`{{#link-to 'index' id='the-link'}}Index{{/link-to}}`));
-  bootApplication();
+    this.visit('/').then(() => {
+      this.$('#the-link').click();
+      let indexController = this.applicationInstance.lookup('controller:index');
 
-  run(jQuery('#the-link'), 'click');
-  let indexController = container.lookup('controller:index');
-  deepEqual(indexController.getProperties('foo', 'bar'), { foo: '123', bar: 'abc' }, 'controller QP properties not');
-});
+      deepEqual(indexController.getProperties('foo', 'bar'),
+        this.indexProperties,
+        'controller QP properties do not update'
+      );
+    });
+  }
 
-QUnit.test('doesn\'t update controller QP properties on current route when invoked (empty query-params obj)', function() {
-  setTemplate('index', compile(`{{#link-to 'index' (query-params) id='the-link'}}Index{{/link-to}}`));
-  bootApplication();
+  ['@test doesn\'t update controller QP properties on current route when invoked (empty query-params obj, inferred route)']() {
+    this.addTemplate('index', `
+      {{#link-to (query-params) id='the-link'}}Index{{/link-to}}
+    `);
 
-  run(jQuery('#the-link'), 'click');
-  let indexController = container.lookup('controller:index');
-  deepEqual(indexController.getProperties('foo', 'bar'), { foo: '123', bar: 'abc' }, 'controller QP properties not');
-});
+    this.visit('/').then(() => {
+      this.$('#the-link').click();
+      let indexController = this.applicationInstance.lookup('controller:index');
 
-QUnit.test('link-to with no params throws', function() {
-  setTemplate('index', compile(`{{#link-to id='the-link'}}Index{{/link-to}}`));
-  expectAssertion(() => bootApplication(), /one or more/);
-});
+      deepEqual(indexController.getProperties('foo', 'bar'),
+        this.indexProperties,
+        'controller QP properties do not update'
+      );
+    });
+  }
+
+  ['@test updates controller QP properties on current route when invoked']() {
+    this.addTemplate('index', `
+      {{#link-to 'index' (query-params foo='456') id="the-link"}}
+        Index
+      {{/link-to}}
+    `);
+
+    this.visit('/').then(() => {
+      run(() => this.$('#the-link').click());
```

Comment:
What I found is that `run` was needed to get the tests to pass here. Why I think it is that way: we are waiting for the query params to change on the controller as a result of user interaction. Is there another way to handle this without waiting via run loop?